### PR TITLE
Add validations for reserved app names when creating new applications.

### PR DESCRIPTION
**Pull request by nipunsampath**
### Related Issues
- https://github.com/wso2-enterprise/asgardeo-product/issues/13467

### Related PRs
- https://github.com/wso2-enterprise/asgardeo-app-extensions/pull/2213

### Checklist
- [ ] e2e cypress tests locally verified (In progress).
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by thivi — APPROVED** [review]

File: apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx [hunk 1]
```diff
@@ -655,6 +672,82 @@ export const MinimalAppCreateWizard: FunctionComponent<MinimalApplicationCreateW
...
+            );
+        }
+    };
+
+    /**
+     * Checks whether the application name is reserved.
+     * @param name - Name of the application
+     */
+    const isAppNameReserved = (name: string) => {
+        if(!reservedAppPattern){
```
Comment (wenodpathirana) [inline apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx:743]: `if (!reservedAppPattern) {`
Add spaces

File: apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx [hunk 2]
```diff
@@ -655,6 +672,82 @@ export const MinimalAppCreateWizard: FunctionComponent<MinimalApplicationCreateW
...
+                t("console:develop.features.applications.forms." +
+                    "spaProtocolSettingsWizard.fields.name.validations.invalid", {
+                    appName: value.toString(),
+                    characterLimit: ApplicationManagementConstants.FORM_FIELD_CONSTRAINTS.APP_NAME_MAX_LENGTH
+                })
+            );
+
+            return;
+        }
+        if (isAppNameReserved(appName)){
```
Comment (wenodpathirana) [inline apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx:695]: `if (isAppNameReserved(appName)) {`

File: apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx [hunk 3]
```diff
@@ -655,6 +672,82 @@ export const MinimalAppCreateWizard: FunctionComponent<MinimalApplicationCreateW
...
+                    "fetchApplications.genericError.message")
+            }));
+        }
+
+        if (response?.applications?.length > 0) {
+            validation.isValid = false;
+            validation.errorMessages.push(
+                t("console:develop.features.applications.forms.generalDetails.fields.name.validations.duplicate")
+            );
+        }
```
Comment (wenodpathirana) [inline apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx:735]: Can't we have these statements within the try `{}` ?

File: apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx (hunk 1 above)
Comment (nipunsampath) [reply apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx:743]: Fixed with 19af1fe13b344dbcdcf0c978195222bdd38d10dc

File: apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx (hunk 2 above)
Comment (nipunsampath) [reply apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx:695]: Fixed with 19af1fe13b344dbcdcf0c978195222bdd38d10dc

File: apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx (hunk 3 above)
Comment (nipunsampath) [reply apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx:735]: Fixed with 19af1fe13b344dbcdcf0c978195222bdd38d10dc

**Pull request merged**